src/sciwrappers.c

```diff
@@ -696,23 +696,24 @@ gint sci_get_lexer(ScintillaObject *sci)
 }
 
 
-void sci_set_lexer(ScintillaObject *sci, guint lexer_id)
+void sci_set_lexer(ScintillaObject *sci, const gchar *lexer_name)
 {
 	gint old = sci_get_lexer(sci);
 
-	/* TODO, LexerNameFromID() is already deprecated */
-	const char *lexer_name = LexerNameFromID(lexer_id);
-	if (! lexer_name)
+	g_return_if_fail(lexer_name != NULL);
+
+	old = sci_get_lexer(sci);
+
+	ILexer5 *lexer = CreateLexer(lexer_name);
+	if (! lexer)
 	{
-		g_warning("Failed to find lexer for ID %u", lexer_id);
+		g_warning("Failed to find lexer for name %s", lexer_name);
 		return;
 	}
 
-	ILexer5 *lexer = CreateLexer(lexer_name);
-
 	SSM(sci, SCI_SETILEXER, 0, (uintptr_t) lexer);
 
-	if (old != (gint)lexer_id)
+	if (old != sci_get_lexer(sci))
```

Comment:
We should probably deprecate `sci_get_lexer()` and stop using it internally.

Reply:
I'm not sure what is the recommendation on this: the [documentation](https://scintilla.org/ScintillaDoc.html#SCI_SETILEXER) says to use this for checking whether setting the new lexer worked.  OTOH, there is [SCI_GETLEXERLANGUAGE](https://scintilla.org/ScintillaDoc.html#SCI_GETLEXERLANGUAGE) which probably would work as well (albeit a bit more expansive).

Reply:
Must be out-dated. Clearly, because of

> Some lexers may not have a lexer ID, just a lexer name in which case 0 is returned.

`SCI_GETLEXER` cannot be reliably used anymore, if using any recent lexer anyway.

Reply:
OK.  I can easily use this here (seems to work), but porting every use in Geany is gonna be more tricky, especially for cases like `highlighting_is_*_style()`

Reply:
Well we'll have to start somewhere. Other places can be fixed incrementally I would say.